pyrsistent/_pmap.py

```diff
@@ -140,7 +140,9 @@ def __str__(self):
 
     def __hash__(self):
         # This hashing algorithm is probably not the speediest
-        return hash(frozenset(self.iteritems()))
+        if not hasattr(self, '_cached_hash'):
```

Comment:
:question: Our PClass hash is called ``__pclass_hash`` it'd be nice to be consistent.